src/lib.rs

```diff
@@ -72,10 +74,27 @@ pub use crate::{
 /// # }
 /// ```
 pub struct LockFile {
+    pub truncate_on_close: bool,
```

Comment:
I'd rather not have this as a public field, it feels like we'd lose control over it. I mean, having a public field like this commits the struct to a specific representation. I'd prefer a private field, and then two methods: `truncate_on_close(&mut self, truncate: bool)` and `truncates_on_close(&self) -> bool`.